src/backend/access/transam/twophase.c

```diff
@@ -1549,8 +1551,32 @@ FinishPreparedTransaction(const char *gid, bool isCommit)
 									   hdr->nabortrels, abortrels,
 									   gid);
 
+	/*
+	 * CSNSnapshot callbacks that should be called right before we are
+	 * going to become visible. Details in comments to this functions.
+	 */
+	if (!iscommit)
```

Comment:
this should be `isCommit`

Reply:
I think this is correct, we are just marking the transaction as aborted in the SLRU.